src/orion/core/io/experiment_builder.py

```diff
@@ -269,23 +271,30 @@ def build_from_config(self, config):
         return experiment
 
     def setup_database(self, config):
-        """Create the Database instance from a configuration.
+        """Create the storage instance from a configuration.
+
+        Note
+        ----
+        Deprecated API, but it is still used heavily in unittests see `setup_protocol`
+
+        """
+        self.setup_protocol(config)
+
+    def setup_protocol(self, config):
+        """Create the storage instance from a configuration.
 
         Parameters
         ----------
         config: dict
             Configuration for the database.
 
         """
-        db_opts = config['database']
+        db_opts = config.get('protocol', {'type': 'legacy'})
```

Comment:
Should add a TODO: Fix this in config refactoring. We are introducing default values here while this should be in `resolve_config`.